lib/resolvers.js

```diff
@@ -29,3 +30,17 @@ export const mapLinkList = fieldName => doc => {
     return linkItem;
   }).filter(l => l !== undefined);
 };
+
+export const mapScheduleList = fieldName => doc => {
+  const schedule = doc[fieldName];
+
+  if (!schedule) { return []; }
+  return schedule.map((item) => {
```

Comment:
> ...I would expect Chris' to actually be less efficient then the original one...
I'm not sure which example you're referring to, @lpil; your one or the one within this PR?

If you were referring to @AndrewBestbier's example, then my motivation was to, remove a superfluous O(n) cycle, and to make it more syntactically elegant.

If however, you're referring to your own example, given the common pattern of using an immutable accumulator in conjunction with a functor, I would _assume_ that the JS engine will be optimised to deal with such transient changes; in this case, efficient storage of indexes.